core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/StaticResourcesEndpointProvider.java

```diff
@@ -27,16 +30,25 @@
 public final class StaticResourcesEndpointProvider implements ResourcesEndpointProvider {
     private final URI endpoint;
     private final Map<String, String> headers;
+    private final Optional<Duration> connectionTimeout;
```

Comment:
Optional should not be used as fields. Too bad this test is ignored for now...

https://github.com/aws/aws-sdk-java-v2/blob/master/test/architecture-tests/src/test/java/software/amazon/awssdk/archtests/CodingConventionTest.java#L87-L91

Reply:
My bad ,  missed it , thanks for the reference 👍